homeassistant/components/sensibo/climate.py

```diff
@@ -148,11 +149,24 @@ def __init__(self, client, data, units):
         """
         self._client = client
         self._id = data["id"]
+        self._productmodel = data["productModel"]
+        self._firmware = data["firmwareVersion"]
+        self._firmwaretype = data["firmwareType"]
         self._external_state = None
         self._units = units
         self._available = False
         self._do_update(data)
         self._failed_update = False
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._id)},
+            name=self._name,
+            manufacturer="Sensibo",
+            configuration_url="https://home.sensibo.com/",
+            model=self._productmodel,
+            sw_version=self._firmware,
+            hw_version=self._firmwaretype,
+            suggested_area=self._name,
```

Comment:
`self._name` doesn't exist at this point.

Reply:
It does as it comes from the `self._do_update(data)` line above. That function sets the name and some other things and also used by the update function.
Anyway I want to clean this whole integration up but first just get the DeviceInfo into it.

Just pushed a small cleanup now so it's ready to be approved (again 😄 )